flocker/node/_docker.py

```diff
@@ -527,6 +568,34 @@ def _address_in_use(self, apierror):
             return None
         return AddressInUse(address=(ip, int(port)), apierror=apierror)
 
+    def _image_data(self, image, container, state):
```

Comment:
A general note on thread-safety: we have potentially multiple threads reading and writing to the dictionary. This is OK, but only because:
1. The data is for our purposes immutable, so two writes of same key isn't a problem.
2. We never remove entries.

You may wish to document these constraints.